### ISSUE TITLE
test_empty_size_pickles crashes

### ISSUE THREAD

**Issue by Whissi** [sabnzbd/sabyenc#18]
Test crashes Python interpreter:
```
tests/test_decoder.py::test_empty_size_pickles Fatal Python error: Aborted

Current thread 0x00007f89ceaea740 (most recent call first):
  File "/var/tmp/portage/dev-python/sabyenc-4.0.1/work/sabyenc-4.0.1/tests/testsupport.py", line 74 in sabyenc3_wrapper
  File "/var/tmp/portage/dev-python/sabyenc-4.0.1/work/sabyenc-4.0.1/tests/test_decoder.py", line 119 in test_empty_size_pickles
  File "/usr/lib64/python3.6/site-packages/_pytest/python.py", line 167 in pytest_pyfunc_call
  File "/usr/lib64/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib64/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib64/python3.6/site-packages/_pytest/python.py", line 1445 in runtest
  File "/usr/lib64/python3.6/site-packages/_pytest/runner.py", line 134 in pytest_runtest_call
  File "/usr/lib64/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib64/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib64/python3.6/site-packages/_pytest/runner.py", line 210 in <lambda>
  File "/usr/lib64/python3.6/site-packages/_pytest/runner.py", line 237 in from_call
  File "/usr/lib64/python3.6/site-packages/_pytest/runner.py", line 210 in call_runtest_hook
  File "/usr/lib64/python3.6/site-packages/_pytest/runner.py", line 185 in call_and_report
  File "/usr/lib64/python3.6/site-packages/_pytest/runner.py", line 99 in runtestprotocol
  File "/usr/lib64/python3.6/site-packages/_pytest/runner.py", line 84 in pytest_runtest_protocol
  File "/usr/lib64/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib64/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib64/python3.6/site-packages/_pytest/main.py", line 271 in pytest_runtestloop
  File "/usr/lib64/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib64/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib64/python3.6/site-packages/_pytest/main.py", line 247 in _main
  File "/usr/lib64/python3.6/site-packages/_pytest/main.py", line 197 in wrap_session
  File "/usr/lib64/python3.6/site-packages/_pytest/main.py", line 240 in pytest_cmdline_main
  File "/usr/lib64/python3.6/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/usr/lib64/python3.6/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib64/python3.6/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib64/python3.6/site-packages/_pytest/config/__init__.py", line 93 in main
  File "/usr/lib/python-exec/python3.6/pytest", line 11 in <module>
/var/tmp/portage/dev-python/sabyenc-4.0.1/temp/environment: line 2783:    66 Aborted                 (core dumped) pytest -vv
```

[build.txt](https://github.com/sabnzbd/sabyenc/files/4313482/build.txt) with gdb backtrace.

**Comment by Safihre** [comment]
Sorry for slow response!
I couldn't reproduce this, also on Travis this error is not thrown on any of the platforms.
https://travis-ci.org/github/sabnzbd/sabyenc/builds/609619999
Could you try with a different python version maybe?

**Comment by mgorny** [comment]
I can reproduce it here. If I disable pytest output capture, I can get a proper assertion message:

```
$ PYTHONPATH=. pytest -s
=============================================================== test session starts ===============================================================
platform linux -- Python 3.9.0b5, pytest-6.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /tmp/sabyenc, configfile: pytest.ini
plugins: asyncio-0.14.0, forked-1.3.0, xdist-1.34.0, hypothesis-5.23.9, mock-3.2.0, flaky-3.7.0, timeout-1.4.2, freezegun-0.4.2
collected 25 items                                                                                                                                

tests/test_decoder.py .............python3.9: src/sabyenc3.c:596: decode_usenet_chunks: Assertion `PyByteArray_Check(PyList_GetItem(Py_input_list, lp))' failed.
Fatal Python error: Aborted

Current thread 0x00007fa8be2ed740 (most recent call first):
  File "/tmp/sabyenc/tests/testsupport.py", line 73 in sabyenc3_wrapper
  File "/tmp/sabyenc/tests/test_decoder.py", line 117 in test_empty_size_pickles
  File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 180 in pytest_pyfunc_call
  File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 1567 in runtest
  File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 153 in pytest_runtest_call
  File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 247 in <lambda>
  File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 294 in from_call
  File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 246 in call_runtest_hook
  File "/usr/lib/python3.9/site-packages/flaky/flaky_pytest_plugin.py", line 138 in call_and_report
  File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 117 in runtestprotocol
  File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 100 in pytest_runtest_protocol
  File "/usr/lib/python3.9/site-packages/flaky/flaky_pytest_plugin.py", line 94 in pytest_runtest_protocol
  File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 321 in pytest_runtestloop
  File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 296 in _main
  File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 240 in wrap_session
  File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 289 in pytest_cmdline_main
  File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
  File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 157 in main
  File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 180 in console_main
  File "/usr/lib/python-exec/python3.9/pytest", line 11 in <module>
Aborted (core dumped)
```

You can't reproduce because Ubuntu defaults to building everything with `-DNDEBUG`, and therefore strips all assertions.

You might be able to reproduce by building it with:
```
$ CFLAGS='-O0 -g' python setup.py build build_ext -i
$ PYTHONPATH=. pytest -s
```

**Comment by Safihre** [comment]
Seems fixed by 5.0.
